lib/wasabi/parser.rb

```diff
@@ -160,19 +160,33 @@ def parse_operations
 
     def parse_types
       schemas.each do |schema|
-        schema.element_children.each do |node|
-
-          case node.name
-          when 'element'
-            complex_type = node.at_xpath('./xs:complexType', 'xs' => XSD)
-            process_type schema, complex_type, node['name'].to_s if complex_type
-          when 'complexType'
-            process_type schema, node, node['name'].to_s
-          end
+        parse_schema(schema)
+      end
+    end
+
+    def parse_schema(schema)
+      schema.element_children.each do |node|
+        case node.name
+        when 'import'
+          url = node.attribute('schemaLocation')
+          process_import url
+        when 'element'
+          complex_type = node.at_xpath('./xs:complexType', 'xs' => XSD)
+          process_type schema, complex_type, node['name'].to_s if complex_type
+        when 'complexType'
+          process_type schema, node, node['name'].to_s
         end
       end
     end
 
+    def process_import(url)
+      xml = Resolver.new(url).resolve
+      document = Nokogiri::XML(xml)
+
+      parse_schema(document.root)
+    rescue StandardError
```

Comment:
Why?

Reply:
in case of any error during loading of external schema - just skip it
specs contains wsdl with import, however referenced .xsd is not in the repository and it was removed from the site also

Reply:
Is it just StandardError or some specific TimeoutSomethingError?

Reply:
there are several cases with different errors
```
case document
        when URL then load_from_remote
        when XML then document
        else          load_from_disc
      end
```